x-pack/plugins/security/server/authentication/providers/base.mock.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { stub, createStubInstance } from 'sinon';
+import { Tokens } from '../tokens';
+
+export function mockAuthenticationProviderOptions() {
+  return {
+    client: { callAsInternalUser: stub(), asScoped: stub(), close: stub() },
+    logger: {
```

Comment:
Should cover cases for `Logger` and `BasePath`:
```ts
import { loggingServiceMock, httpServiceMock } from '../../../../../../src/core/server/mocks';

export function mockAuthenticationProviderOptions() {
  const logger = loggingServiceMock.create();
  const basePath = httpServiceMock.createSetupContract().basePath;
  basePath.get.mockReturnValue('/base-path');

  return {
    client: { callAsInternalUser: stub(), asScoped: stub(), close: stub() },
    logger,
    basePath,
    tokens: createStubInstance(Tokens),
  };
}
```
I will add mocks for ClusterClient, ScopedClusterClient and KibanaRequest in a separate PR

Reply:
Awesome!